src/views/SignMessage.vue

```diff
@@ -12,97 +26,152 @@
 
 <script lang="ts">
 import { Component, Emit, Vue } from 'vue-property-decorator';
-import { State } from 'vuex-class';
-import { SmallPage } from '@nimiq/vue-components';
+import { State, Getter } from 'vuex-class';
+import { SmallPage, AccountSelector } from '@nimiq/vue-components';
 import { RequestType, ParsedSignMessageRequest } from '../lib/RequestTypes';
 import staticStore, { Static } from '../lib/StaticStore';
 import { WalletStore } from '@/lib/WalletStore';
 import { AccountInfo } from '@/lib/AccountInfo';
 import { WalletInfo } from '@/lib/WalletInfo';
-import {
-    // @ts-ignore
-    SignMessageRequest as KSignMessageRequest, SignMessageResult as KSignMessageResult,
-} from '@nimiq/keyguard-client';
+import KeyguardClient from '@nimiq/keyguard-client';
 import { Utf8Tools } from '@nimiq/utils';
 import { ERROR_CANCELED } from '@/lib/Constants';
+import { State as RpcState } from '@nimiq/rpc';
 
-@Component({components: {SmallPage}})
+@Component({components: {SmallPage, AccountSelector}})
 export default class SignMessage extends Vue {
     @Static protected request!: ParsedSignMessageRequest;
-    @State private keyguardResult!: KSignMessageResult;
+    @Static private rpcState!: RpcState;
+    @State private keyguardResult!: KeyguardClient.SignMessageResult;
+    @Getter private processedWallets!: WalletInfo[];
+    @Getter private findWallet!: (id: string) => WalletInfo | undefined;
+
+    private showAccountSelector = false;
+
+    private async created() {
+        if (this.keyguardResult) return;
+
+        if (this.request.walletId && this.request.signer) {
+            this.accountSelected(this.request.walletId, this.request.signer.toUserFriendlyAddress());
+            return;
+        }
+
+        this.showAccountSelector = true;
+
+    }
 
-    protected sendKeyguardRequest(walletInfo: WalletInfo, accountInfo: AccountInfo) {
-        const request: KSignMessageRequest = {
+    private async accountSelected(walletId: string, address: string) {
```

Comment:
Maybe `selectAccount` is a better name.

Reply:
It's literally the function that get assigned as the handler for the `account-selected` event of the account selector. I'd rather name it `onAccountSelected` then. Is that what you mean?

Reply:
That doesn't really fit, because it's also called in `created()` if the request contains the signer.